### PR TITLE
verifier: fix issuers with ports

### PR DESCRIPTION
**Pull request by calebdoxsey**
Currently when using automatic JWKS verification we parse issuers using code that doesn't support alternative port numbers (it ends up treating the domain name as a scheme). This PR fixes that so both `example.com` and `example.com:1234` will work.

It also changes the go-jose pkg to v2, which we use everywhere else.

### CODE REVIEW

File: sdk.go [hunk 1]
```diff
@@ -195,3 +183,32 @@ func (v *Verifier) fetchJWKSFromRemote(ctx context.Context, u, keyID string) (*j
...
+func (v *Verifier) getVerifyEndpoint(tok *jwt.JSONWebToken) (string, error) {
+	verifyEndpoint := v.staticJWKSEndpoint
+	if verifyEndpoint == "" {
+		out := jwt.Claims{}
+		if err := tok.UnsafeClaimsWithoutVerification(&out); err != nil {
+			return "", fmt.Errorf("couldn't get json web key: %w", err)
+		}
+
+		rawURL := out.Issuer
+		if !strings.HasPrefix(rawURL, "https://") && !strings.HasPrefix(rawURL, "http://") {
```
Comment (wasaga) [inline sdk.go:196]: we parse this URL just below anyway, you can compare a Scheme of URL object instead maybe?

File: sdk.go [hunk 2]
```diff
@@ -195,3 +183,32 @@ func (v *Verifier) fetchJWKSFromRemote(ctx context.Context, u, keyID string) (*j
...
+			rawURL = "https://" + rawURL
+		}
+
+		u, err := url.Parse(rawURL)
+		if err != nil {
+			return "", fmt.Errorf("invalid issuer in JWT: %s", out.Issuer)
+		}
+		if u.Scheme == "" {
```
Comment (wasaga) [inline sdk.go:204]: duplicate of the above strings.HasPrefix

**Review by wasaga — CHANGES_REQUESTED** [review]

File: sdk.go (hunk 1 above)
Comment (calebdoxsey) [reply sdk.go:196]: that's how it worked before. It ended up parsing `example.com:1234` as a url with a scheme of `example.com`.

Comment (wasaga) [reply sdk.go:196]: TIL

**Pull request merged**